samples/condor/boinc_gahp_protocol.txt

```diff
@@ -0,0 +1,916 @@
+    subsequent line of output with. This may simplify parsing the output of
+    the GAHP server by the client program, especially in cases where
+    the responses of more than one GAHP server are "collated" together.
+
+    This affects the output of both return lines and result lines for all
+    subsequent commands (NOT including the current one).
+
+    + Request Line:
+
+        RESPONSE_PREFIX <SP> <prefix> <CRLF>
+
+        <prefix> = an arbitrary string of characters which you want to prefix
+        every subsequent line printed by the GAHP server with.
+
+    + Return Line:
+
+        S <CRLF>
+
+    + Result Line:
+
+        None.
+
+    + Example:
+
+        S: RESPONSE_PREFIX BOINC-GAHP:
+        R: S
+        S: RESULTS
+        R: BOINC-GAHP:S 0
+        S: RESPONSE_PREFIX NEW_PREFIX_
+        R: BOINC-GAHP:S
+        S: RESULTS
+        R: NEW_PREFIX_S 0
+
+    -----------------------------------------------
+
+    QUIT
+
+    Free any/all system resources (close all sockets, etc) and terminate
+    as quickly as possible.
+
+    + Request Line:
+
+        QUIT <CRLF>
+
+    + Return Line:
+
+        S <CRLF>
+
+        Immediately afterwards, the command pipe should be closed
+        and the BOINC-GAHP server should terminate.
+
+    + Result Line:
+
+        None.
+
+    -----------------------------------------------
+
+    RESULTS
+
+    Display all of the Result Lines which have been queued since the
+    last RESULTS command was issued.  Upon success, the first return
+    line specifies the number of subsequent Result Lines which will be
+    displayed.  Then each result line appears (one per line) -- each
+    starts with the request ID which corresponds to the request ID
+    supplied when the corresponding command was submitted.  The exact
+    format of the Result Line varies based upon which corresponding
+    Request command was issued.
+
+    IMPORTANT: Result Lines must be displayed in the _exact order_ in
+    which they were queued!!!  In other words, the Result Lines
+    displayed must be sorted in the order by which they were placed into
+    the BOINC-GAHP's result line queue, from earliest to most recent.
+
+    + Request Line:
+
+        RESULTS <CRLF>
+
+    + Return Line(s):
+
+        S <SP> <num-of-subsequent-result-lines> <CRLF>
+        <reqid> <SP> ... <CRLF>
+        <reqid> <SP> ... <CRLF>
+        ...
+
+        * reqid = integer Request ID, set to the value specified in
+            the corresponding Request Line.
+
+    + Result Line:
+
+        None.
+
+    + Example:
+
+        S: RESULTS
+        R: S 1
+        R: 100 NULL
+
+    -----------------------------------------------
+
+    3.5.2 GAHP COMMANDS SPECIFIC TO BOINC
+
+    The following commands are specific to the BOINC-GAHP.
+
+    -----------------------------------------------
+
+    BOINC_SELECT_PROJECT
+
+    Select a project to be used for subsequent commands. This is a synchronous
+    command even when in asynchronous mode!
```

Comment:
The BOINC-GAHP server will only have a default if read for the config file, which I think was implemented for easier testing. Either we should completely specify the config file (name, location, contents) or be completely silent about it in the specification. IMHO the config file is part of the GAHP daemon implementation, not of the protocol. Conformant to the protocol not having specified a BOINC project should result in an error, even if a config file is present.